packages/core/src/conflict-resolution.ts

```diff
@@ -255,40 +247,37 @@ export class ConflictResolution {
    */
   private async applyLogToState<T extends Stream>(
     handler: StreamHandler<T>,
-    log: LogEntry[],
+    unappliedCommits: CommitData[],
     state?: StreamState,
     breakOnAnchor?: boolean
   ): Promise<StreamState> {
     // When we have genesis state only, and add some commits on top, we should check a signature at particular timestamp.
     // Most probably there is a timestamp information there. If no timestamp found, we consider it to be _now_.
     // `fetchLog` provides the timestamps.
     if (state && state.log.length === 1) {
-      const timestamp = log[0].timestamp
+      const timestamp = unappliedCommits[0].timestamp
       const genesisCid = state.log[0].cid
       const genesis = await this.dispatcher.retrieveCommit(genesisCid)
       await handler.applyCommit(genesis, { cid: genesisCid, timestamp: timestamp }, this.context)
     }
-    const itr = log.entries()
+    const itr = unappliedCommits.entries()
     let entry = itr.next()
     while (!entry.done) {
-      const logEntry = entry.value[1]
-      const commit = await this.dispatcher.retrieveCommit(logEntry.cid)
+      const logEntry = await this.getCommitData(entry.value[1])
+      const commitMeta = { cid: logEntry.cid, timestamp: logEntry.timestamp }
       // TODO - should catch potential thrown error here
 
-      let payload = commit
-      if (StreamUtils.isSignedCommit(commit)) {
-        payload = await this.dispatcher.retrieveCommit(commit.link)
-      }
-
-      if (payload.proof) {
-        // it's an anchor commit
+      if (StreamUtils.isAnchorCommit(logEntry.commit)) {
+        // It's an anchor commit
         // TODO: Anchor validation should be done by the StreamHandler as part of applying the anchor commit
-        await verifyAnchorCommit(this.dispatcher, this.anchorValidator, commit)
-        state = await handler.applyCommit(commit, logEntry, this.context, state)
+        await verifyAnchorCommit(this.dispatcher, this.anchorValidator, logEntry.commit)
+        state = await handler.applyCommit(logEntry.commit, commitMeta, this.context, state)
       } else {
-        // it's a signed commit
-        const tmpState = await handler.applyCommit(commit, logEntry, this.context, state)
-        const isGenesis = !payload.prev
+        // It's a signed commit but may not be using DagJWS. If the JWS envelope is present, use that while applying the
+        // commit, otherwise use the commit itself (e.g. for CAIP-10 links).
+        const commitToApply = logEntry.envelope ? logEntry.envelope : logEntry.commit
```

Comment:
I assume this will go away with the follow-up PR that makes the handler utilize the CommitData more directly?

Reply:
Did we end up creating a ticket that tracks this?

Reply:
I did, yes, #1532

Reply:
And to answer your question, @stbrody, yes, this will go away.